packages/outsystems-wrapper/src/index.ts

```diff
@@ -0,0 +1,124 @@
+import { OSGLOCPosition, PluginError, Position, PositionOptions } from "../../cordova-plugin/src/definitions";
+
+class OSGeolocation {
+    #lastPosition: Position | null = null
+
+    getCurrentPosition(success: (position: Position) => void, error: (err: PluginError | GeolocationPositionError) => void, options: PositionOptions): void {
+        // @ts-ignore
+        if (typeof (cordova) === 'undefined' && typeof (CapacitorUtils) === 'undefined') {
+            // if we're not in cordova / capacitor land, we call the good old Web API
+            navigator.geolocation.getCurrentPosition(success, error, options);
+            return
+        }
+
+        let timeoutID: ReturnType<typeof setTimeout> | undefined;
+        const successCallback = (position: Position | OSGLOCPosition) => {
+            if (typeof (timeoutID) == 'undefined') {
+                // Timeout already happened, or native fired error callback for
+                // this geo request.
+                // Don't continue with success callback.
+                return;
+            }
+
+            if (this.#isLegacyPosition(position)) {
+                position = this.#convertFromLegacy(position)
+            }
+            clearTimeout(timeoutID)
+
+            this.#lastPosition = position
+            success(position)
+        }
+
+        const errorCallback = (e: PluginError) => {
+            if (typeof (timeoutID) !== 'undefined') {
+                clearTimeout(timeoutID)
+            }
+            error(e)
+        }
+
+        // Check our cached position, if its timestamp difference with current time is less than the maximumAge, then just
+        // fire the success callback with the cached position.
+        if (this.#lastPosition && options.maximumAge && (((new Date()).getTime() - this.#lastPosition.timestamp) <= options.maximumAge)) {
+            success(this.#lastPosition);
```

Comment:
I see this was taken from how the cordova plugin was setup - but On Android, this logic is being managed by the actual [FusedLocationProviderClient](https://developers.google.com/android/reference/com/google/android/gms/location/FusedLocationProviderClient). I don't know how it is on iOS, but if there is analagous logic, perhaps there is no need to store the `lastPosition` in this wrapper.

Reply:
I checked with @OS-ricardomoreirasilva and iOS does not have this logic, so it needs to exist here